h2o-core/src/main/java/water/rapids/ast/prims/string/AstReplaceAll.java

```diff
@@ -65,16 +66,29 @@ public Val apply(Env env, Env.StackHelp stk, AstRoot asts[]) {
   }
 
   private Vec replaceAllCategoricalCol(Vec vec, String pattern, String replacement, boolean ignoreCase) {
+    final Pattern compiledPattern = Pattern.compile(pattern); // Compile the pattern once before replacement
     String[] doms = vec.domain().clone();
-    for (int i = 0; i < doms.length; ++i)
+    Set<String> newDomainSet = new HashSet<>(); // The pattern might create multiple domains with the same name
```

Comment:
Might get big, but is fast to search for duplicated. Not sure if there is better alternative without sacrificing speed.

Reply:
IMHO not an issue.